lib/robots/dor_repo/gis_delivery/load_geoserver.rb

```diff
@@ -214,39 +247,56 @@ def create_raster(catalog, ws, layer)
+              )
+              LyberCore::Log.debug "load-geoserver: #{druid} loaded style #{raster_style}"
+              if style_exists.nil?
+                LyberCore::Log.debug "load-geoserver: #{druid} saving new style #{raster_style}"
+                style.create(style_name: raster_style)
+                style.update(style_name: raster_style, filename: nil, payload: sldtxt)
               end
             else
               raster_style = 'raster_grayband' # a simple band-oriented histogram adjusted style
-              style = RGeoServer::Style.new catalog, name: raster_style
-              fail "load-geoserver: #{druid} has missing style #{raster_style} on #{catalog}" if style.new?
+              style_exists = style.find(
+                style_name: raster_style
+              )
+              fail "load-geoserver: #{druid} has missing style #{raster_style}" if style_exists.nil?
             end
           else
-            style = RGeoServer::Style.new catalog, name: raster_style
-            fail "load-geoserver: #{druid} has missing style #{raster_style} on #{catalog}" if style.new?
+            style_exists = style.find(
+              style_name: raster_style
+            )
+            fail "load-geoserver: #{druid} has missing style #{raster_style}" if style_exists.nil?
           end
 
           # fetch layer to load raster style - it's created when the coverage is created via REST API
-          lyr = RGeoServer::Layer.new catalog, workspace: ws, name: druid
-          if lyr.new?
-            fail "load-geoserver: Layer #{druid} is missing for coverage #{ws.name}/#{cs.name}/#{druid}"
+          layer = Geoserver::Publish::Layer.new(connection)
+          layer_exists = layer.find(layer_name: druid)
+          if layer_exists.nil?
+            fail "load-geoserver: Layer #{druid} is missing for coverage #{workspace_name}/#{druid}/#{druid}"
           end
 
-          if lyr.default_style != style.name
-            lyr.default_style = style.name
-            LyberCore::Log.debug "load-geoserver: #{druid} updating #{lyr.name} with default style #{style.name}"
+          if layer_exists.dig('layer', 'defaultStyle', 'name') != raster_style
+            layer_exists['layer']['defaultStyle'] = raster_style
+            LyberCore::Log.debug "load-geoserver: #{druid} updating #{druid} with default style #{raster_style}"
             begin
-              lyr.save
-            rescue RGeoServer::GeoServerInvalidRequest => e
+              layer.update(layer_name: druid, additional_payload: layer_exists)
+            rescue Geoserver::Publish::Error => e
               fail "load-geoserver: #{druid} cannot save Layer: #{e.message}"
             end
           end
         end
+
+        def determine_rights(druid)
+          rights = 'Restricted'
+          xml = Dor.find("druid:#{druid}").rightsMetadata.ng_xml
+          if xml.search('//rightsMetadata/access[@type=\'read\']/machine/world').length > 0
```

Comment:
```suggestion
if Dor.find("druid:#{druid}").rights == 'World'
```

Reply:
Would you mind if I fix this up on a follow on PR? I think there will be some specs that might fail.